tools/clang/lib/SPIRV/SPIRVEmitter.cpp

```diff
@@ -2816,10 +2822,11 @@ uint32_t SPIRVEmitter::processGetSamplePosition(const CXXMemberCallExpr *expr) {
   const auto sampleCount = theBuilder.createUnaryOp(
       spv::Op::OpImageQuerySamples, theBuilder.getUint32Type(),
       loadIfGLValue(object));
-  emitWarning(
-      "GetSamplePosition only supports standard sample settings with 1, 2, 4, "
-      "8, or 16 samples and will return float2(0, 0) for other cases",
-      expr->getCallee()->getExprLoc());
+  if (!spirvOptions.noWarnEmulatedFeatures)
+    emitWarning("GetSamplePosition only supports standard sample settings with "
+                "1, 2, 4, "
+                "8, or 16 samples and will return float2(0, 0) for other cases",
+                expr->getCallee()->getExprLoc());
```

Comment:
This warning text doesn't tell me anything about the fact that the generated SPIR-V will emulate this feature.

Maybe the warning could say:
GetSamplePosition is emulated using many SPIR-V instructions due to the lack of a direct SPIR-V equivalent.
The emulated GetSamplePosition only supports standard sample settings with 1, 2, 4, 8, or 16 samples, and will return float2(0,0) for other cases.

Reply:
Done. Thanks for the suggestion!